File: code-external/fib.go
package main
import (
    "fmt"
    "net/http"
    "time"

    "github.com/tealeg/xlsx" // Excel file package
)

func handler(w http.ResponseWriter, r *http.Request) {
    // Your handler logic here
    fmt.Fprintf(w, "Hello, World!")
}

func main() {
    http.HandleFunc("/", handler)
    go http.ListenAndServe(":8080", nil)

    // Create a new Excel file
    var file *xlsx.File
    var sheet *xlsx.Sheet
    var row *xlsx.Row
    var cell *xlsx.Cell

    file = xlsx.NewFile()
    sheet, err := file.AddSheet("Request Times")
    if err != nil {
        fmt.Printf("Failed to create sheet: %v", err)
        return
    }

    for i := 0; i < 1000; i++ {
        start := time.Now()
        resp, err := http.Get("http://localhost:8080")
        if err != nil {
            fmt.Printf("Request failed: %v", err)
            continue
        }
        resp.Body.Close()
        elapsed := time.Since(start)

        // Write the time taken for each request to the Excel sheet in milliseconds with 3 decimal places
        row = sheet.AddRow()
        cell = row.AddCell()
        cell.SetValue(fmt.Sprintf("%.3f", float64(elapsed)/float64(time.Millisecond)))
    }

    // Save the Excel file
    err = file.Save("RequestTimes.xlsx")
    if err != nil {
        fmt.Printf("Failed to save Excel file: %v", err)
    }
}

File: rust/src/fib.rs
// use std::time::Instant;
// use warp::Filter;
// use reqwest::get;
// use xlsxwriter::*;
// #[tokio::main]
// async fn main() {
//     let route = warp::path!("hello" / String)
//         .map(|name| {
//             format!("Hello, {}!", name)
//         });
 
//     tokio::spawn(async move {
//         warp::serve(route).run(([127, 0, 0, 1], 8080)).await;
//     });
 
//     // Create a new workbook
//     let workbook = Workbook::new("RequestTimes.xlsx");
//     let mut sheet = workbook.add_worksheet(None).unwrap();
 
//     // Send 1000 requests to the server
//     for i in 0..1000 {
//         let start = Instant::now();
//         let resp = get("http://localhost:8080/hello/world").await;
//         match resp {
//             Ok(_) => {
//                 let elapsed = start.elapsed();
//                 // Write the time taken for each request to the Excel sheet
//                 sheet.write_string(i as u32, 0, &format!("{:?}", elapsed), None).unwrap();
//             }
//             Err(err) => println!("Request failed: {:?}", err),
//         }
//     }
 
//     // Save the workbook
//     workbook.close().unwrap();
// }
#[derive(Deserialize)]
struct YourRequestBodyType {
    key: String,
}

use warp::Filter;
use serde::Deserialize;

#[derive(Deserialize)]
struct YourRequestBodyType {
    key: String,
}

#[tokio::main]
async fn main() {
    let get_route = warp::path("your_get_endpoint")
        .and(warp::get())
        .map(|| warp::reply::json(&"Response for GET request"));

    let post_route = warp::path("your_post_endpoint")
        .and(warp::post())
        .and(warp::body::json())
        .map(|body: YourRequestBodyType| {
            warp::reply::json(&format!("Received POST request with body: {:?}", body))
        });

    let routes = get_route.or(post_route);
    warp::serve(routes).run(([127, 0, 0, 1], 3030)).await;
}
